projects/graph/src/graph.py

```diff
@@ -6,4 +6,19 @@
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
-        pass  # TODO
+        self.vertices = {}
+        
+    def add_vertex(self, vertex):
+        if vertex not in self.vertices:
+            self.vertices[vertex] = set()
+
+    def add_edge(self, vertex_1, vertex_2):
+        if vertex_1 in self.vertices and vertex_2 in self.vertices:
+            self.vertices[vertex_1].add(vertex_2)
+            self.vertices[vertex_2].add(vertex_1)
+        else:
+            if vertex_1 not in self.vertices and vertex_2 not in self.vertices:
+                raise KeyError(f"{vertex_1} and {vertex_2} are not valid vertices")
```

Comment:
Great error checking you specify which vertex is not in the graph, Samantha, and this is all O(1)